### PR TITLE
feat: create dynfed main fn

### PR DESCRIPTION
**Pull request by kobayu858**
- Create dynfed main fn
- Explicitly states that the cycle and the deadline are equal.
- For PR #58

### CODE REVIEW

File: 2021_RTCSA_dynfed/src/main.rs [hunk 1]
```diff
@@ -1,2 +1,54 @@
...
+fn main() {
+    let arg: ArgParser = ArgParser::parse();
+    let mut dag_set = create_dag_set_from_dir(&arg.dag_dir_path);
+
+    for dag in dag_set.iter_mut() {
+        if let (Some(period), Some(end_to_end_deadline)) =
+            (dag.get_head_period(), dag.get_end_to_end_deadline())
+        {
+            assert_eq!(
+                end_to_end_deadline, period,
```
Comment (atsushi421) [inline 2021_RTCSA_dynfed/src/main.rs:42]: It may be better to just give a warning and set end_to_end_deadline to the same value as the period.

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: 2021_RTCSA_dynfed/src/main.rs (hunk 1 above)
Comment (kobayu858) [reply 2021_RTCSA_dynfed/src/main.rs]: Change error to warning and overwrite end_to_end_deadline with period.

File: 2021_RTCSA_dynfed/src/main.rs [hunk 2]
```diff
@@ -1,2 +1,61 @@
...
+fn main() {
+    let arg: ArgParser = ArgParser::parse();
+    let mut dag_set = create_dag_set_from_dir(&arg.dag_dir_path);
+
+    for dag in dag_set.iter_mut() {
+        if let (Some(period), Some(end_to_end_deadline)) =
+            (dag.get_head_period(), dag.get_end_to_end_deadline())
+        {
+            if end_to_end_deadline != period {
+                warn!("period and end_to_end_deadline must be equal, Override end_to_end_deadline with period.");
```
Comment (atsushi421) [inline 2021_RTCSA_dynfed/src/main.rs:43]:
```suggestion
                warn!("In this algorithm, the period and the end-to-end deadline must be equal. Therefore, the end-to-end deadline is overridden by the period.");
```

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: 2021_RTCSA_dynfed/src/main.rs (hunk 2 above)
Comment (kobayu858) [reply 2021_RTCSA_dynfed/src/main.rs]: Grammar corrections.

**Review by atsushi421 — APPROVED** [review]
LGTM

**Pull request merged**